pytools/__init__.py

```diff
@@ -1486,30 +1486,106 @@ def __call__(self, *args, **kwargs):
 class Table:
     """An ASCII table generator.
 
+    :arg alignments: List of alignments of each column ('l', 'c', or 'r',
+        for left, center, and right alignment, respectively). Columns which
+        have no alignment specifier will use the last specified alignment. For
+        example, with `alignments=['l', 'r']`, the third and all following
+        columns will use 'r' alignment.
+
     .. automethod:: add_row
     .. automethod:: __str__
     .. automethod:: latex
+    .. automethod:: github_markdown
     """
 
-    def __init__(self):
+    def __init__(self, alignments=None):
         self.rows = []
+        if alignments is not None:
+            self.alignments = alignments
+        else:
+            self.alignments = ['l']
 
     def add_row(self, row):
         self.rows.append([str(i) for i in row])
 
     def __str__(self):
+        """
+        Returns a string representation of the table.
+
+        .. doctest ::
+
+            >>> tbl = Table(alignments=['l', 'r', 'l'])
+            >>> tbl.add_row([1, '|'])
+            >>> tbl.add_row([10, '20||'])
+            >>> print(tbl)
+            1  |    |
+            ---+------
+            10 | 20||
```

Comment:
Is the duplicated `|` intentional here?

Reply:
Yes, the test input matches the one in `github_markdown` and is supposed to illustrate the difference in the output between the two functions.

Reply:
OIC. I had missed the pipe characters in the input.